lib/processors/resourceListCreator.js

```diff
@@ -0,0 +1,646 @@
+				// assign theme resources additionally to theme packages
+				for (const [prefix, list] of this._themePackages.entries()) {
+					if ( resource.name.startsWith(prefix) ) {
+						list.add(resource);
+						contained = true;
+					}
+				}
+			}
+
+			if ( contained ) {
+				this._resources.delete(resource.name);
+			}
+		}
+	}
+
+	get resources() {
+		return new Set(this._resources.keys());
+	}
+
+	get components() {
+		return this._components;
+	}
+
+	get themePackages() {
+		return this._themePackages;
+	}
+}
+
+/**
+ * Whether the detection of orphans should result in a build failure.
+ *
+ * @since 1.29.1
+ */
+
+module.exports = async function({resources, dependencyResources}, options) {
+	options = Object.assign({
+		failOnOrphans: true,
+		externalResources: undefined,
+		debugResources: DEFAULT_DEBUG_RESOURCES_FILTER.join(","),
+		mergedResources: DEFAULT_BUNDLE_RESOURCES_FILTER.join(","),
+		designtimeResources: DEFAULT_DESIGNTIME_RESOURCES_FILTER.join(","),
+		supportResources: DEFAULT_SUPPORT_RESOURCES_FILTER.join(",")
+	}, options);
+
+	const pool = new LocatorResourcePool();
+	await pool.prepare( resources.concat(dependencyResources) );
+
+	const collector = new ResourceCollector(pool);
+	resources.forEach((resource) => collector.visitResource(resource.getPath()));
+	log.info("	found %d resources", collector.resources.size);
+
+	// determine additional information for the found resources
+	if ( options && options.externalResources ) {
+		collector.setExternalResources(options.externalResources);
+	}
+	await collector.determineResourceDetails({
+		pool,
+		debugResources: options.debugResources,
+		mergedResources: options.mergedResources,
+		designtimeResources: options.designtimeResources,
+		supportResources: options.supportResources
+	});
+
+	// group resources by components and create ResourcesLists
+	collector.groupResourcesByComponents();
+
+	const resourceLists = [];
+
+	// write out resources.json files
+	for (const [prefix, list] of collector.components.entries()) {
+		log.info(`	writing '${prefix}resources.json'`);
+		resourceLists.push(new EvoResource({
+			path: `/resources/${prefix}resources.json`,
+			string: JSON.stringify(list, null, "\t")
+		}));
+	}
+	for (const [prefix, list] of collector.themePackages.entries()) {
+		log.info(`	writing '${prefix}resources.json'`);
+		resourceLists.push(new EvoResource({
+			path: `/resources/${prefix}resources.json`,
+			string: JSON.stringify(list, null, "\t")
+		}));
+	}
+	const unassigned = collector.resources;
+	if ( unassigned.size > 0 ) {
+		log.warn(`  found ${unassigned.size} resources not belonging to a component (orphans)`);
+		let n = 0;
+		for ( const resource of unassigned ) {
+			log.verbose(`    ${resource} (orphan)`);
+			if ( ++n > 20 ) {
+				log.verbose("		... (%d more)", unassigned.size - n);
+				break;
+			}
+		}
+		if ( options.failOnOrphans ) {
+			throw new Error("not all resources could be assigned to components");
+		}
+	}
```

Comment:
Is there a guideline where errors should be thrown, processors or tasks?

Reply:
I think the lowest level is here perfect because we have all the information there and typically the error should bubble all the way up

Reply:
That's fine, but the error message should try to explain the context better. The call stack might get lost through catching and wrapping this error while bubbling up.

Some good examples:
* https://github.com/SAP/ui5-builder/blob/master/lib/processors/uglifier.js#L27
* https://github.com/SAP/ui5-builder/blob/master/lib/processors/bundlers/flexChangesBundler.js#L66

Reply:
done